compiler/src/main/scala/org/scalajs/nscplugin/GenJSExports.scala

```diff
@@ -723,8 +728,17 @@ trait GenJSExports[G <: Global with Singleton] extends SubComponent {
           s"found overloaded default getter $defaultGetter")
 
       val trgTree = {
-        if (sym.isClassConstructor || static) genLoadModule(trgSym)
-        else js.This()(encodeClassType(trgSym))
+        if (sym.isClassConstructor || static) {
+          assert(trgSym.isLifted == captures.nonEmpty,
+              "inconsistent lifted symbol / captures")
+
+          if (!trgSym.isLifted)
+            genLoadModule(trgSym)
+          else
+            genNew(trgSym, trgSym.primaryConstructor, captures)
```

Comment:
I'm pretty sure this is incorrect. We could create several instances of the module per enclosing instance, if we use a `new` every time. We should use the *getter* method on the enclosing instance instead. If the enclosing instance is a JS class, that means accessing the property of its `JSName`; if it is a Scala class, it means calling the Scala method with its name and 0 argument.

Reply:
We should add a side-effect in the constructor of the companion object in the tests, and test that it is only applied once per enclosing instance.